scripts/dts/extract/heuristics.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) 2018 Bobby Noelte
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from extract.globals import *
+from extract.edts import *
+from extract.directive import DTDirective
+from extract.default import default
+import pprint
+
+##
+# @brief Generate device tree information based on heuristics.
+#
+# Generates in EDTS:
+# - bus/master : device id of bus master for a bus device
+# - parent-device : device id of parent device
+class DTHeuristics(DTDirective):
```

Comment:
After the change of the extract function signature - why do you need to derive the class from DTDirective?

Reply:
Fixed